File: frontend/src/types/student.ts
export interface Student {
  id: string;
  studentId: string;
  userId: string;
  email: string;
  firstName: string;
  lastName: string;
  dateOfBirth: string;
  major?: string;
  yearLevel?: string;
  enrollmentDate: string;
  graduationDate?: string;
  phoneNumber?: string;
  address?: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateStudentRequest {
  studentId: string;
  userId: string;
  dateOfBirth: string;
  major?: string;
  yearLevel?: string;
  enrollmentDate: string;
  graduationDate?: string;
  phoneNumber?: string;
  address?: string;
  status?: string;
}

export interface UpdateStudentRequest {
  dateOfBirth?: string;
  major?: string;
  yearLevel?: string;
  enrollmentDate?: string;
  graduationDate?: string;
  phoneNumber?: string;
  address?: string;
  status?: string;
}

File: backend/src/main/java/com/example/starter_project_2025/system/student/entity/Student.java
package com.example.starter_project_2025.system.student.entity;

import com.example.starter_project_2025.system.user.entity.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "students")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(unique = true, nullable = false, length = 50)
    private String studentId;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(nullable = false)
    private LocalDate dateOfBirth;

    @Column(length = 100)
    private String major;

    @Column(length = 50)
    private String yearLevel;

    @Column(nullable = false)
    private LocalDate enrollmentDate;

    @Column
    private LocalDate graduationDate;

    @Column(length = 15)
    private String phoneNumber;

    @Column(length = 500)
    private String address;

    @Column(nullable = false)
    private String status = "ACTIVE";

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
